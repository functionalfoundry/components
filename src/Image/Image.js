/* @flow */
import React from 'react'
import Theme from 'js-theme'
import mergeProps from 'js-theme/lib/mergeProps'

type PropsT = {
  alt: string,
  testId?: string,
  theme: Object,
}

const defaultProps = {
  theme: {},
}

const Image = ({
  alt,
  testId, // eslint-disable-line no-unused-vars
  theme,
  ...props,
}: PropsT) => (
  <img
    alt={alt}
    {...mergeProps(theme.image, props)}
  />
)

Image.defaultProps = defaultProps

const defaultTheme = {
  image: {
    borderWidth: 0,
    borderStyle: 'solid',
    boxSizing: 'border-box',
    margin: 0,
    padding: 0,
    minHeight: 0,
    minWidth: 0,
  },
}

export default Theme('Image', defaultTheme)(Image)
