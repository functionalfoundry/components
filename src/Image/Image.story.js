import React from 'react'
import { storiesOf } from '@kadira/storybook'
import Image from './Image'

storiesOf('Image', module)
  .add('Regular', () => (
    <div
      style={styles.container}
    >
      <Image
        src='https://pbs.twimg.com/profile_images/657199367556866048/EBEIl2ol.jpg'
      />
    </div>
  ))
  .add('Fixed width', () => (
    <div
      style={styles.container}
    >
      <Image
        src='https://pbs.twimg.com/profile_images/657199367556866048/EBEIl2ol.jpg'
        width={100}
        height={100}
      />
    </div>
  ))

const styles = {
  container: {
    width: '100%',
  },
}
