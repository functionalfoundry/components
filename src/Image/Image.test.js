import React from 'react'
import renderer from 'react-test-renderer'
import Image from '.'

test('Renders without props', () => {
  const view = renderer
    .create(
      <Image
        src="https://pbs.twimg.com/profile_images/657199367556866048/EBEIl2ol.jpg"
        width={100}
        height={100}
      />
    )
    .toJSON()
  expect(view).toMatchSnapshot()
})
