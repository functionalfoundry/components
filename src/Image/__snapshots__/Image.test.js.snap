exports[`test Renders without props 1`] = `
<img
  alt={undefined}
  className="image_xsbxd7"
  height={100}
  src="https://pbs.twimg.com/profile_images/657199367556866048/EBEIl2ol.jpg"
  width={100} />
`;
